sdk/eventhub/Azure.Messaging.EventHubs/src/Processor/EventProcessor.cs

```diff
@@ -607,5 +643,35 @@ private Task RenewOwnershipAsync()
 
             return Manager.ClaimOwnershipAsync(ownershipToRenew);
         }
+
+        /// <summary>
+        ///   Invokes a specified action only if this <see cref="EventProcessor" /> instance is not running.
+        /// </summary>
+        ///
+        /// <param name="action">The action to invoke.</param>
+        ///
+        /// <exception cref="InvalidOperationException">Occurs when this method is invoked while the event processor is running.</exception>
+        ///
+        private void AssertNotRunning(Action action)
```

Comment:
nit:  We may want to avoid "Assert" here, as it is used around the code for "check this condition and throw if it isn't right" where this actually performs an action.   Maybe something like `EnsureNotRunningAndInvoke`?   Tricky name here.

Reply:
I'll stick to that.